### PR TITLE
drivers: sam: add DDR self-refresh for sama7g5 low-power support

### PR DESCRIPTION
**Pull request by TonyHan11**

### CODE REVIEW

File: core/drivers/pm/sam/pm_suspend.S [hunk 1]
```diff
@@ -327,12 +327,29 @@ SUSPEND_FUNC(at91_pm_suspend_in_sram)
 	mov	tmp1, #0
 	mcr	p15, 0, tmp1, c7, c10, 4
 
+	/* Flush tlb. */
+	mov	tmp1, #0
+	mcr	p15, 0, tmp1, c8, c7, 0
```
Comment (jenswi-linaro) [inline core/drivers/pm/sam/pm_suspend.S:332]: Please use the `write_tlbiall` macro from the generated `out/core/include/generated/arm32_sysreg.S` instead.
Do you need an `isb` after this?

File: core/drivers/pm/sam/at91_pm.c [hunk 2]
```diff
@@ -373,15 +373,42 @@ TEE_Result atmel_pm_suspend(uintptr_t entry, struct sm_nsec_ctx *nsec)
...
+#endif
+	};
+	uint32_t i = 0;
 	int node = -1;
 	size_t size = 0;
 
-	node = fdt_node_offset_by_compatible(fdt, -1, "atmel,sama5d3-ddramc");
-	if (node < 0)
-		return TEE_ERROR_ITEM_NOT_FOUND;
+	for (; i < ARRAY_SIZE(dram_map); i++) {
```
[the remark this answers is not in the document]
Comment (TonyHan11) [reply core/drivers/pm/sam/at91_pm.c]: Updated, thank you for your suggestion.

File: core/drivers/pm/sam/pm_suspend.S (hunk 1 above)
Comment (TonyHan11) [reply core/drivers/pm/sam/pm_suspend.S]: Updated, thanks.

**Review by jenswi-linaro — COMMENTED** [review]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by TonyHan11** [comment]
Updated with review tags added and rebased on the latest `master`, thanks.

**Pull request merged**